Remove unused function

matklad opened the pull request.

matklad committed:
Remove unused function

rust-highfive commented:
r? @nikomatsakis

(rust_highfive has picked a reviewer for you, use r? to override)

matklad commented:
r? @petrochenkov

petrochenkov commented:
@bors r+ rollup

bors commented:
:pushpin: Commit 080fb8b8c63b4e61ba052138b0b7d75571aff317 has been approved by `petrochenkov`

<!-- @bors r=petrochenkov 080fb8b8c63b4e61ba052138b0b7d75571aff317 -->
<!-- homu: {"type":"Approved","sha":"080fb8b8c63b4e61ba052138b0b7d75571aff317","approver":"petrochenkov"} -->

matklad commented:
bors r-

Used in tests, that's why it wasn't caught by x.py check

matklad commented:
@bors r-

matklad committed:
Move lexer unit tests to rustc_lexer

StringReader is an intornal abstraction which at the moment changes a
lot, so these unit tests cause quite a bit of friction.

Moving them to rustc_lexer and more ingerated-testing style should
make them much less annoying, hopefully without decreasing their
usefulness much.

Note that coloncolon tests are removed (it's unclear what those are
testing).

\r\n tests are removed as well, as we normalize line endings even
before lexing.

matklad commented:
Moved the tests, ready for review

petrochenkov commented on src/librustc_lexer/src/tests.rs:
```diff
@@ -128,6 +128,11 @@ fn check_lexing(src: &str, expect: Expect) {
 
+#[test]
+fn smoke_test() {
+    check_lexing("/* my source file */ fn main() { println!(\"zebra\"); }\n", expect![[]])
```
>expect![[]]

The test output wasn't updated?

matklad replied:
Ah indeed... At least now we now that these tests are actually capable of failing on CI:

https://github.com/rust-lang/rust/runs/1044897833?check_suite_focus=true#step:23:2341

petrochenkov commented:
r=me with https://github.com/rust-lang/rust/pull/76050#discussion_r479636065 addressed

matklad commented:
@bors r=petrochenkov

bors commented:
:pushpin: Commit 5da8bf9501b18d2c4db2f81fec106bfc93e9d9f3 has been approved by `petrochenkov`

<!-- @bors r=petrochenkov 5da8bf9501b18d2c4db2f81fec106bfc93e9d9f3 -->
<!-- homu: {"type":"Approved","sha":"5da8bf9501b18d2c4db2f81fec106bfc93e9d9f3","approver":"petrochenkov"} -->

matklad commented:
Rebased over the great compiler renaming

@bors r=petrochenkov

bors commented:
:pushpin: Commit 518cac91902d34567ac8bfea3022f426a7de53f6 has been approved by `petrochenkov`

<!-- @bors r=petrochenkov 518cac91902d34567ac8bfea3022f426a7de53f6 -->
<!-- homu: {"type":"Approved","sha":"518cac91902d34567ac8bfea3022f426a7de53f6","approver":"petrochenkov"} -->
